packages/flutter/test/widgets/navigator_test.dart

```diff
@@ -544,6 +544,39 @@ void main() {
     await tester.pumpAndSettle();
   });
 
+  testWidgets('Pop no animation page does not crash', (WidgetTester tester) async {
+    // Regression Test for https://github.com/flutter/flutter/issues/86604.
+    Widget buildNavigator(bool secondPage) {
+      return Navigator(
+        pages: <Page<void>>[
+          const ZeroDurationPage(
+            child: Text('page1'),
+          ),
+          if (secondPage)
+            const ZeroDurationPage(
+              child: Text('page2'),
+            ),
+        ],
+        onPopPage: (Route<dynamic> route, dynamic result) => false,
+      );
+    }
+    await tester.pumpWidget(
+      Directionality(
```

Comment:
ubter-nit: for slightly cleaner code, put the Directionality widget inside buildNavigator?